code/modules/mob/living/simple_animal/friendly/pet.dm

```diff
@@ -23,6 +23,16 @@
 	if(P.tagname && !unique_pet)
 		fully_replace_character_name(null, "\proper [P.tagname]")
 
+/mob/living/simple_animal/pet/Move(NewLoc, direct)
+	. = ..()
+
+	if(pcollar)
+		if(mobility_flags)
+			if(loc == NewLoc)
+				if(!has_gravity(loc))
+					return
+				pcollar.step_action()
```

Comment:
Early returns are generally prefered, also whats up with the mobility_flags there?
```suggestion
	if(!pcollar)
		return
	if (!(mobility_flags & MOBILITY_STAND))
		return
	if(loc != NewLoc)
		return
	if(!has_gravity(loc))
		return
		
	pcollar.step_action()
```